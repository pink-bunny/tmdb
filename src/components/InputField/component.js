import React from 'react';
import PropTypes from 'prop-types';
import { getIn } from 'formik';
import { Form, Input, Icon } from 'antd';

const InputField = ({
  placeholder,
  autoComplete,
  type,
  icon,
  field,
  form: { touched, errors },
  ...props
}) => {
  const fieldTouched = getIn(touched, field.name);
  const fieldErrors = getIn(errors, field.name);

  return (
    <Form.Item
      validateStatus={fieldErrors && fieldTouched && 'error'}
      help={fieldErrors && fieldTouched && fieldErrors}
    >
      <Input
        type={type}
        autoComplete={autoComplete}
        placeholder={placeholder}
        prefix={(
          icon && (
            <Icon
              type={icon}
              style={{ color: 'rgba(0,0,0,.25)' }}
            />
          )
        )}
        {...field}
        {...props}
      />
    </Form.Item>
  );
};

InputField.propTypes = {
  placeholder: PropTypes.string,
  field: PropTypes.shape().isRequired,
  form: PropTypes.shape({
    touched: PropTypes.shape().isRequired,
    errors: PropTypes.shape().isRequired
  }).isRequired,
  autoComplete: PropTypes.string,
  type: PropTypes.string,
  icon: PropTypes.string
};
InputField.defaultProps = {
  placeholder: '',
  type: 'text',
  icon: null,
  autoComplete: 'off'
};

export default InputField;
