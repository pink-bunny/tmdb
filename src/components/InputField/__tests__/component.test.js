import React from 'react';
import { shallow } from 'enzyme';

import InputField from '../index';

describe('InputField component matches snapshot', () => {
  const fieldName = 'password';
  const requiredProps = {
    field: {
      name: fieldName
    },
    form: {
      touched: {},
      errors: {}
    }
  };

  it('with required props', () => {
    const wrapper = shallow(<InputField {...requiredProps} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('with icon', () => {
    const props = {
      ...requiredProps,
      icon: 'lock'
    };
    const wrapper = shallow(<InputField {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('with custom props', () => {
    const props = {
      ...requiredProps,
      type: 'password',
      autoComplete: 'on',
      placeholder: 'Enter password',
      icon: 'lock'
    };
    const wrapper = shallow(<InputField {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('when field is touched and has error', () => {
    const props = {
      ...requiredProps,
      form: {
        touched: {
          [fieldName]: true
        },
        errors: {
          [fieldName]: 'Reqired'
        }
      }
    };
    const wrapper = shallow(<InputField {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });
});
