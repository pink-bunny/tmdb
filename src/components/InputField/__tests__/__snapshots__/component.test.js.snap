// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component matches snapshot when field is touched and has error 1`] = `
<FormItem
  hasFeedback={false}
  help="Reqired"
  validateStatus="error"
>
  <Input
    autoComplete="off"
    name="password"
    placeholder=""
    prefix={null}
    type="text"
  />
</FormItem>
`;

exports[`InputField component matches snapshot with custom props 1`] = `
<FormItem
  hasFeedback={false}
  help=""
>
  <Input
    autoComplete="on"
    name="password"
    placeholder="Enter password"
    prefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="lock"
      />
    }
    type="password"
  />
</FormItem>
`;

exports[`InputField component matches snapshot with icon 1`] = `
<FormItem
  hasFeedback={false}
  help=""
>
  <Input
    autoComplete="off"
    name="password"
    placeholder=""
    prefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="lock"
      />
    }
    type="text"
  />
</FormItem>
`;

exports[`InputField component matches snapshot with required props 1`] = `
<FormItem
  hasFeedback={false}
  help=""
>
  <Input
    autoComplete="off"
    name="password"
    placeholder=""
    prefix={null}
    type="text"
  />
</FormItem>
`;
